### PR TITLE
Adds Error handling for provider config deserialization

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: provider/provider_config/snowflake_config.go [hunk 1]
```diff
@@ -54,7 +56,7 @@ func (sf *SnowflakeConfig) HasLegacyCredentials() bool {
 
 func (sf *SnowflakeConfig) HasCurrentCredentials() (bool, error) {
 	if (sf.Account != "" && sf.Organization == "") || (sf.Account == "" && sf.Organization != "") {
-		return false, fmt.Errorf("credentials must include both Account and Organization")
+		return false, fferr.NewProviderConfigError(provider_type.SnowflakeOffline, fmt.Errorf("credentials must include both Account and Organization"))
```
Comment (epps) [inline provider/provider_config/snowflake_config.go:59]: Looks like you'll need to cast to a string here.

File: provider/provider_config/snowflake_config.go [hunk 2]
```diff
@@ -120,16 +122,16 @@ func (sf *SnowflakeConfig) getBaseConnection() (string, error) {
 	isLegacy := sf.HasLegacyCredentials()
 	isCurrent, err := sf.HasCurrentCredentials()
 	if err != nil {
-		return "", fmt.Errorf("could not check credentials: %v", err)
+		return "", err
 	}
 	if isLegacy && isCurrent {
-		return "", fmt.Errorf("cannot use both legacy and current credentials")
+		return "", fferr.NewProviderConfigError(provider_type.SnowflakeOffline, fmt.Errorf("cannot use both legacy and current credentials"))
```
Comment (epps) [inline provider/provider_config/snowflake_config.go:128]: Cast to string here.

File: provider/provider_config/snowflake_config.go [hunk 3]
```diff
@@ -120,16 +122,16 @@ func (sf *SnowflakeConfig) getBaseConnection() (string, error) {
...
 	} else if isLegacy && !isCurrent {
 		return fmt.Sprintf("%s:%s@%s/%s/%s", sf.Username, sf.Password, sf.AccountLocator, sf.Database, sf.schema()), nil
 	} else if !isLegacy && isCurrent {
 		return fmt.Sprintf("%s:%s@%s-%s/%s/%s", sf.Username, sf.Password, sf.Organization, sf.Account, sf.Database, sf.schema()), nil
 	} else {
-		return "", fmt.Errorf("credentials not found")
+		return "", fferr.NewProviderConfigError(provider_type.SnowflakeOffline, fmt.Errorf("credentials not found"))
```
Comment (epps) [inline provider/provider_config/snowflake_config.go:134]: Cast to string here.

File: provider/provider_config/k8s_config.go [hunk 4]
```diff
@@ -54,13 +55,13 @@ func (k8s *K8sConfig) UnmarshalJSON(data []byte) error {
 	} else {
 		err = k8s.decodeExecutor(temp.ExecutorType, temp.ExecutorConfig)
 		if err != nil {
-			return fmt.Errorf("could not decode executor: %w", err)
+			return err
```
Comment (anthonylasso) [inline provider/provider_config/k8s_config.go:58]: missing fferr here?

File: provider/provider_config/k8s_config.go [hunk 5]
```diff
@@ -54,13 +55,13 @@ func (k8s *K8sConfig) UnmarshalJSON(data []byte) error {
...
 		}
 	}
 
 	err = k8s.decodeFileStore(temp.StoreType, temp.StoreConfig)
 	if err != nil {
-		return fmt.Errorf("could not decode filestore: %w", err)
+		return err
```
Comment (anthonylasso) [inline provider/provider_config/k8s_config.go:64]: here as well?

Comment (sdreyer) [reply provider/provider_config/k8s_config.go:64]: We wrap it within that function

File: provider/provider_config/k8s_config.go (hunk 4 above)
Comment (sdreyer) [reply provider/provider_config/k8s_config.go:58]: wrapped within that function

**Pull request merged**